Meta/Lagom/Tools/CodeGenerators/LibTextCodec/GenerateEncodingIndexes.cpp

```diff
@@ -0,0 +1,224 @@
+    return {};
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    StringView generated_header_path;
+    StringView generated_implementation_path;
+    StringView json_path;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(generated_header_path, "Path to the lookup table header file to generate", "generated-header-path", 'h', "generated-header-path");
+    args_parser.add_option(generated_implementation_path, "Path to the lookup table implementation file to generate", "generated-implementation-path", 'c', "generated-implementation-path");
+    args_parser.add_option(json_path, "Path to the JSON file to read from", "json-path", 'j', "json-path");
+    args_parser.parse(arguments);
+
+    auto json_file = TRY(Core::File::open(json_path, Core::File::OpenMode::Read));
+    auto json_data = TRY(json_file->read_until_eof());
+    auto data = TRY(JsonValue::from_string(json_data)).as_object();
+
+    auto gb18030_table = prepare_table(data.get("gb18030"sv)->as_array());
+
+    // FIXME: Encoding specification is not updated to GB-18030-2022 yet (https://github.com/whatwg/encoding/issues/312)
+    // NOTE: See https://commits.webkit.org/264918@main
+    gb18030_table.code_points[7182] = 0xfe10;
+    gb18030_table.code_points[7183] = 0xfe12;
+    gb18030_table.code_points[7184] = 0xfe11;
+    gb18030_table.code_points[7185] = 0xfe13;
+    gb18030_table.code_points[7186] = 0xfe14;
+    gb18030_table.code_points[7187] = 0xfe15;
+    gb18030_table.code_points[7188] = 0xfe16;
+    gb18030_table.code_points[7201] = 0xfe17;
+    gb18030_table.code_points[7202] = 0xfe18;
+    gb18030_table.code_points[7208] = 0xfe19;
+    gb18030_table.code_points[23775] = 0x9fb4;
+    gb18030_table.code_points[23783] = 0x9fb5;
+    gb18030_table.code_points[23788] = 0x9fb6;
+    gb18030_table.code_points[23789] = 0x9fb7;
+    gb18030_table.code_points[23795] = 0x9fb8;
+    gb18030_table.code_points[23812] = 0x9fb9;
+    gb18030_table.code_points[23829] = 0x9fba;
+    gb18030_table.code_points[23845] = 0x9fbb;
+
+    LookupTables tables {
+        .gb18030_ranges = data.get("gb18030-ranges"sv)->as_array(),
+        .indexes = {
+            { "gb18030"sv, move(gb18030_table) },
+            { "big5"sv, prepare_table(data.get("big5"sv)->as_array()) },
+            { "jis0208"sv, prepare_table(data.get("jis0208"sv)->as_array()) },
+            { "jis0212"sv, prepare_table(data.get("jis0212"sv)->as_array()) },
+            { "euc_kr"sv, prepare_table(data.get("euc-kr"sv)->as_array()) },
+        },
+    };
+
+    auto generated_header_file = TRY(Core::File::open(generated_header_path, Core::File::OpenMode::Write));
+    auto generated_implementation_file = TRY(Core::File::open(generated_implementation_path, Core::File::OpenMode::Write));
+
+    TRY(generate_header_file(tables, *generated_header_file));
+    TRY(generate_implementation_file(tables, *generated_implementation_file));
```

Comment:
we should try to only write these files to the final location if they have changed. ref IDLGenerators.cpp

Reply:
I basically copied GenerateGLAPIWrapper here.
After randomly sampling a couple of the other code generators, it looks like almost all of them do it like this.
IDLGenerators is the only one that doesn't unconditionally write to the output files.

Besides, `invoke_generator` already does `cmake -E copy_if_different foo.tmp foo`.
So I don't think the copying the `write_if_changed` implementation is justified here.
Maybe if it was in a library that all code generators could use...

Reply:
Sounds like a future yak for me then, thanks for looking into it :)